src/sinol_make/helpers/compile.py

```diff
@@ -27,6 +70,16 @@ def compile(program, output, compilers: Compilers = None, compile_log = None, we
     if extra_compilation_files is None:
         extra_compilation_files = []
 
+    compiled_exe = check_compiled(program)
+    if compiled_exe is not None:
+        try:
+            compile_log.write(f'Using cached executable {compiled_exe}\n')
+        except AttributeError:
+            pass
```

Comment:
This try-except is very confusing. It looks like it's catching an exception from writing to the log file and not from the log file being None -- it's less readable than it was previously.

Reply:
So I changed it to what it was before